cdap-docs/reference-manual/source/http-restful-api/metrics.rst

```diff
@@ -371,161 +410,193 @@ To query a metric within a given context, perform an HTTP POST request::
+              "grouping":{},
+              "data":[{"time":1429487786,"value":1268},
+                      {"time":1429487787,"value":1324},
+                      {"time":1429487788,"value":1206},
+                      {"time":1429487789,"value":1125},
+                      {"time":1429487790,"value":1035}]},
+             {"metricName":"system.dataset.store.bytes",
+              "grouping":{},
+              "data":[{"time":1429487786,"value":15600},
+                      {"time":1429487787,"value":14998},
+                      {"time":1429487788,"value":13712},
+                      {"time":1429487789,"value":12246},
+                      {"time":1429487790,"value":9924}]
+              }]
+  }
 
-  For example, to request a user-defined metric for the *HelloWorld* Application's *WhoFlow* Flow::
+.. rubric:: Multiple Metrics with Different Contexts
 
-    POST '<base-url>/metrics/query?context=namespace.default.app.HelloWorld.flow.WhoFlow.flowlet.saver
-      &metric=user.names.bytes&aggregate=true'
+Retrieving multiple metrics at once for different contexts can be accomplished by issuing
+a request with a JSON list as the request body that enumerates the name, attributes and
+context for each metric. Use an HTTP POST request:: 
 
-.. _http-restful-api-v3-metrics-multiple:
-.. _http-restful-api-metrics-multiple:
+  POST 'http://<host>:<port>/v3/metrics/query'
 
-Querying for Multiple Metrics
------------------------------
-Retrieving multiple metrics at once, by issuing an HTTP POST request with a JSON list as
-the request body that enumerates the name and attributes for each metric, is currently not
-supported in this API. Instead, use the v2 API until it is supported in a future release.
+with the arguments as a JSON string in the body. The format of the JSON follows this
+structure (pretty-printed)::
 
-To retrieve multiple metrics at once, instead of a GET, issue an HTTP POST, with a JSON
-list as the request body that enumerates the name and attributes for each metrics. For
-example::
+  { “query1”: {
+        tags: {“namespace”: “default”, “app”: “PurchaseHistory”}, 
+        metrics: [“metric1”, “metric2”],
+        groupBy: [“app”, “dataset”],
+        timeRange: {“aggregate”: “true”}
+        },
+    “query2”: {
+        tags: {“namespace”: “default”},
+        metrics: [“metric1”, “metric2”],
+        groupBy: [“app”, “dataset”],
+        timeRange: {“start”: “now­2s”, “end”: “now”}
+        }
+  }
 
-  POST http://<host>:<port>/v2/metrics
+For example, to retrieve multiple metrics using a ``curl`` call (results reformatted to fit)::
 
-with the arguments as a JSON string in the body. For example, to retrieve multiple metrics
-using a ``curl`` call::
+  $ curl -w'\n' -X POST 'http://localhost:10000/v3/metrics/query' -H 'Content-Type: application/json' \
+   -d '{"query1":{"tags": {"flow":"CountRandom"}, "metrics": ["system.process.events.processed"], "timeRange": {"start":"now-5s", "count":"5"}}}'
```

Comment:
since this is an example to demonstrate querying multiple metrics, i would give an example where i can query at least 2 metrics.